src/CPSC559/WorkerThread.java

```diff
@@ -93,9 +104,11 @@ private void processReq(String command, boolean forward) throws IOException {
                 // Command format: s_BookTitle
                 Book b = searchBook(command.split("_")[1]);
                 if (b == null) {
-                    output.println("nack");
+                    String retMsg = "nack%" + command + ';';
+                    output.println(retMsg);
                 } else {
-                    output.println(b.toString());
+                    String retMsg = "ack%" + command + ';' + b.toString();
```

Comment:
You declare this variable a lot in here which causes a compile error. If your just declare it as null at the top of the method and then make all these `retMsg="whatever"` it should work.